app/src/main/java/ru/chaichuk/hwapp/MainActivity.kt

```diff
@@ -1,21 +1,32 @@
 package ru.chaichuk.hwapp
 
-import android.content.Intent
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
-import android.view.View
-import android.widget.Button
+import androidx.appcompat.app.AppCompatActivity
+import ru.chaichuk.hwapp.fragments.MoviesDetailsFragment
 import ru.chaichuk.hwapp.fragments.MoviesListFragment
+import ru.chaichuk.hwapp.fragments.OnMoviesDetailsClickListener
+import ru.chaichuk.hwapp.fragments.OnMoviesListClickListener
 
-class MainActivity : AppCompatActivity() {
+class MainActivity :
+    AppCompatActivity(),
+    OnMoviesListClickListener,
+    OnMoviesDetailsClickListener {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+    }
+
+    override fun onMovieClick() {
+        supportFragmentManager.beginTransaction()
+            .replace(R.id.mainFrame, MoviesDetailsFragment())
+            .addToBackStack(null)
+            .commit()
+    }
 
+    override fun onDetailsBack() {
```

Comment:
no no no 
To come back don't need to create a new instance and add to back stack
just `supportFragmentManager.popBackStack()` enough